cyclic_boosting/base.py

```diff
@@ -23,6 +23,75 @@
 _logger = logging.getLogger(__name__)
 
 
+class UpdateMixin(object):
```

Comment:
These classes were previously at the bottom of the file, which prevented correct typing for `red`